lsp-fiber/src/rpc.ml

```diff
@@ -152,10 +181,16 @@ struct
         let error = Jsonrpc.Response.Error.make ~code ~message () in
         Fiber.return (Reply.now (Response.error req.id error), state)
       | Ok (In_request.E r) -> (
-        let+ response = h_on_request.on_request t r in
+        let cancel = Cancel.create () in
+        Table.set t.pending req.id cancel;
+        let+ response =
+          Fiber.Var.set Cancel.var cancel (fun () ->
+              h_on_request.on_request t r)
+        in
         match response with
         | Error e -> (Reply.now (Response.error req.id e), state)
         | Ok (response, state) ->
+          Cancel.destroy cancel;
```

Comment:
You need to destroy it in case of errors as well, right?

Reply:
Good call. And I also forgot to remove the key from the table. Should be fixed.